### ISSUE TITLE
Inconsistent write-event pattern usage in Governor extensions

### ISSUE THREAD

**Issue by vladyan18** [OpenZeppelin/openzeppelin-contracts#4507]
**📝 Details**

Hello again :) For some reason, in the extension contracts for Governor, pattern "state variable change -> event" is not followed. For example: 

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/21716722ad78ee3e9bf21a5b47ff420323c8fe27/contracts/governance/extensions/GovernorSettings.sol#L85-L88

The event is emitted before the value actually changes. At the same time, it is important to note that in other modules of the library, events are emitted strictly after the changes.

I can assume that such a variant is implemented for the sake of saving gas or minimizing lines of code - so as not to read the storage once again. However, according to my estimates, this **does not save gas**. At least, if the changing value (`_votingDelay` in this case) is packed into the same slot along with some other value, the compiler still has to read the entire slot before repacking the new value into it.

**tldr:** I think it is actually uses more gas, and it's different from the rest of the codebase.

Maybe I'm wrong and there is a reason for this approach, in which case it might make sense to explain it.
